### ISSUE TITLE
Make pop_fetch_t take const char* as param

### ISSUE THREAD

**Issue by fekir** [neomutt/neomutt#1777]
Following callbacks function
````
static int fetch_message(char *line, void *data);
static int fetch_capa(char *line, void *data);
static int fetch_auth(char *line, void *data);
static int check_uidl(char *line, void *data);
````

take `line` as non-const pointer.

On the caller site, the code looks like

````
callback(inbuf, data)
mutt_mem_realloc(&inbuf, lenbuf + sizeof(buf));
//...
FREE(&inbuf);
````

Thus there is the undocumented but not so unexpected assumption that `callback` does not `realloc` or `free` `inbuf`.

If the callback would take an `const char*` as parameter, then such errors could be prevented at compile-time.

Fortunately none of the callbacks functions edits `inbuf`, expect for `check_uidl`, which calls `memmove(line, endp, strlen(endp) + 1);`, and afterwards `mutt_str_strcmp(edata->uid, line) == 0)`.

I'm unsure about why the `memmove` is necessary.
It seems that it copies the leading `\0` (because of `strlen`+1), thus it could be deleted and the second function call replaced with `mutt_str_strcmp(edata->uid, endp) == 0`

If this is the case, the definition of `pop_fetch_t` could be updated to `typedef int (*pop_fetch_t)(const char *str, void *data);`.

**Comment by flatcap** [comment]
> I'm unsure about why the `memmove` is necessary.

I've just debugged the code and I agree, the `memmove` is unneccesary.
`endp` points to the `NUL`-terminated string we want to compare.

Please make the changes.

**Comment by flatcap** [comment]
Fixed by 42c55c16d
